signer/src/storage/postgres.rs

```diff
@@ -736,6 +764,129 @@ impl PgStore {
         .await
         .map_err(Error::SqlxQuery)
     }
+
+    /// Check whether the given block hash is a part of the stacks
+    /// blockchain identified by the given chain-tip.
+    async fn in_canonical_stacks_blockchain(
+        &self,
+        chain_tip: &model::StacksBlockHash,
+        block_hash: &model::StacksBlockHash,
+        block_height: u64,
+    ) -> Result<bool, Error> {
+        sqlx::query_scalar::<_, bool>(
+            r#"
+            WITH RECURSIVE tx_block_chain AS (
+                SELECT
+                    block_hash
+                  , block_height
+                  , parent_hash
+                FROM sbtc_signer.stacks_blocks
+                WHERE block_hash = $1
+
+                UNION ALL
+
+                SELECT
+                    child.block_hash
+                  , child.block_height
+                  , child.parent_hash
+                FROM sbtc_signer.stacks_blocks AS child
+                JOIN tx_block_chain AS parent
+                  ON child.block_hash = parent.parent_hash
+                WHERE parent.block_height > $2
+            )
+            SELECT EXISTS (
+                SELECT TRUE
+                FROM tx_block_chain AS tbc
+                WHERE tbc.block_hash = $3
+            );
+        "#,
+        )
+        .bind(chain_tip)
+        .bind(i64::try_from(block_height).map_err(Error::ConversionDatabaseInt)?)
+        .bind(block_hash)
+        .fetch_one(&self.0)
+        .await
+        .map_err(Error::SqlxQuery)
+    }
+
+    /// Fetch a status summary of a withdrawal request.
+    ///
+    /// In this query we fetch the raw withdrawal request and add some
+    /// information about whether this signer accepted the request.
+    ///
+    /// `None` is returned if withdrawal request is not in the database or
+    /// if the withdrawal request is not associated with a stacks block in
+    /// the database.
+    async fn get_withdrawal_request_status_summary(
+        &self,
+        id: &model::QualifiedRequestId,
+        signer_public_key: &PublicKey,
+    ) -> Result<Option<WithdrawalStatusSummary>, Error> {
+        sqlx::query_as::<_, WithdrawalStatusSummary>(
+            r#"
+            SELECT
+                ws.is_accepted
+              , wr.amount
+              , wr.max_fee
+              , wr.recipient
+              , wr.block_height AS bitcoin_block_height
+              , wr.block_hash   AS stacks_block_hash
+              , sb.block_height AS stacks_block_height
+            FROM sbtc_signer.withdrawal_requests AS wr
+            JOIN sbtc_signer.stacks_blocks AS sb USING (block_hash)
```

Comment:
Just a note: I usually avoid using `USING()` in joins because it can make the query's intent less clear. While it's not an issue here, in more complex queries it can be harder to quickly understand which tables are involved (or intended to be involved) in the join. There's also a risk of accidentally joining columns like `block_hash` when they have multiple meanings in different tables (e.g., Bitcoin vs. Stacks) which can result in unexpected results. To avoid ambiguity and the risk of unintended joins, I'd recommend using explicit `ON` clauses as a default.

Reply:
Great point, I'll update it.

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/1344/commits/14e9f7e1b4c62e0ebb1a156062f2895cc2bd7715